src/pages/upload/index.tsx

```diff
@@ -30,10 +30,10 @@ const fileUploadReducer = (state: any, action: any) => {
 
 const Upload: React.FC = () => {
   const [state, dispatch] = React.useReducer(fileUploadReducer, {files: []})
-  const {viewer} = useViewer()
+  const {ability} = useViewer()
   const uploaderRef = React.useRef(null)
 
-  return viewer?.s3_signing_url ? (
+  return ability.can('upload', 'Video') ? (
```

Comment:
I like how this is coming together. Moving from an indirect presence check of a URL to an ability check of the action taken on this page. This makes it much clearer what is going on.